daemon/mounts.go

```diff
@@ -18,6 +18,10 @@ func (daemon *Daemon) prepareMountPoints(container *container.Container) error {
 		if err := daemon.lazyInitializeVolume(container.ID, config); err != nil {
 			return err
 		}
+		if config.Volume == nil {
```

Comment:
This looks to be the best fix for now; though we should probably come up with a better interface for mountpoints (e.g. a method to check if they're volume-like/convert them into a volume), since they currently feel like a very leaky abstraction.